cmd/podman/pods/create.go

```diff
@@ -274,6 +284,17 @@ func create(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+// MapOptions is necessary for ensuring that createOptions contains all necessary data
+// for usage by the libpod api endpoint. assigns leftover containerCLIopts to podCreateopts
+func MapOptions(infraOptions entities.ContainerCLIOpts, createOpts entities.PodCreateOptions, img string) entities.PodCreateOptions {
+	toReturn := createOpts
+	toReturn.InfraConmonPidFile = infraOptions.ConmonPIDFile
+	toReturn.InfraImage = img
+	toReturn.InfraName = infraOptions.Name
+	toReturn.Hostname = infraOptions.Hostname
```

Comment:
This and Userns feel like they should be in the normal infra config - why are they being handled separately?

Reply:
the `MapOpts` function is solely for the purpose of the libpod api endpoint since I pass a `podcreateoptions` type, I cannot pass two types so I end up re-parsing it on that end. when using `podman-remote` it is easier to pass both infra and pod create but since you can also call it directly through `curl` that is the limiting factor as to what can/should be passed.